util/common.go

```diff
@@ -175,3 +176,25 @@ func Chunks(s string, chunkSize int) []string {
 
 	return chunks
 }
+
+// GenerateCacheKey return cacheKey by query type/class/domain
+func GenerateCacheKey(qType uint16, qClass uint16, qName string) string {
```

Comment:
Class type can be omitted, IN is used in 99.9%, this would save 2 bytes per cache entry